usmqe/usmmail.py

```diff
@@ -0,0 +1,82 @@
+"""
+Module for working with email alerts.
+
+.. moduleauthor:: ebondare
+
+Usage::
+
+    import usmqe.usmmail
+
+    # ...get all  messages from /var/mail/root on the client machine
+    # Return a string
+
+    messages = usmqe.usmmail.get_client_mail()
+
+    # ...get messages from /var/mail/root on the client machine
+    # that came within the specified time interval
+    # Return a mailbox object
+
+    messages = usmqe.usmmail.get_msgs_by_time(start_timestamp, end_timestamp)
+"""
+
+
+import pytest
+import usmqe.usmssh
+import mailbox
+import time
+import email.utils
+import os
+
+
+LOGGER = pytest.get_logger('usmmail', module=True)
+
+
+def get_client_mail():
+    """
+    Read mail from /var/mail/root on client machine.
+    Return the contents of the mailbox as a string
+    """
+    SSH = usmqe.usmssh.get_ssh()
+    host = usmqe.inventory.role2hosts("usm_client")[0]
+    cat_mail_log_cmd = "cat /var/mail/root"
+    retcode, stdout, stderr = SSH[host].run(cat_mail_log_cmd)
+    if retcode != 0:
+        raise OSError(stderr)
+    return stdout.decode()
+
+
+def get_msgs_by_time(start_timestamp=None, end_timestamp=None):
+    """
+    Get all the messages from /var/mail/root on the client machine.
+    Choose the ones that came within the specified time interval.
+
+    Return a mailbox object.
+    """
+    # Pretend we have a Date header; get the date from the Received header
+    client_mail = get_client_mail().replace(';', '\nDate:')
+    with open('mailbox_file', 'w') as f:
+        f.write(client_mail)
+    mailbox_instance = mailbox.mbox('mailbox_file')
+    relevant_messages = mailbox.mbox('relevant_messages_mailbox')
```

Comment:
Would it make sense to use `tmpdir` fixture for storing temporary files like these mailbox files? See https://docs.pytest.org/en/latest/tmpdir.html#base-temporary-directory